### DISCUSSION TITLE
On secret token. Is it actually needed?

### DISCUSSION THREAD

**Discussion by niklavsBariss** [rnmapbox/maps#3699] [General]
During my attempts storing secret (DOWNLOADS:READ) token in a .gitignored project scoped file, I realised everything builds fine without `~/.netrc` or any of these lines
```
            authentication {
                basic(BasicAuthentication)
            }
            credentials {
                // Do not change the username below.
                // This should always be `mapbox` (not your username).
                username = 'mapbox'
                // Use the secret token you stored in gradle.properties as the password
                password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: ""
            }
```
* Is the secret token really needed for _react native_ mapbox? Seems like a legacy requirement no longer needed.
* If it is, is there a way to securely store different secret tokens for different projects?

**Comment by niklavsBariss** [comment]
Turns out `pod install` fails with a wrong token in `.netrc` 
```
[!] Error installing MapboxCommon
[!] /usr/bin/curl -f -L -o /var/folders/0j/4_n4kw2d0b70yntlcld_48sm0000gn/T/d20241121-46140-sm9r09/file.zip https://api.mapbox.com/downloads/v2/mapbox-common/releases/ios/packages/23.11.0/MapboxCommon.zip --create-dirs --netrc-optional --retry 2 -A 'CocoaPods/1.14.3 cocoapods-downloader/2.1'
```

**Comment by shancock884** [comment]
My guess is that if it works without the download token, then it would be because gradle has a cached copy of the library from a previous build.
